crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -142,6 +142,7 @@ pub struct SystemState<Param: SystemParam> {
 impl<Param: SystemParam> SystemState<Param> {
     pub fn new(world: &mut World) -> Self {
         let mut meta = SystemMeta::new::<Param>();
+        meta.last_change_tick = world.last_change_tick;
         let param_state = <Param::Fetch as SystemParamState>::init(world, &mut meta);
```

Comment:
Leaving the `last_change_tick` set to `0` was probably not correct. From what I could tell, initializing a system state "late" would cause it to miss changes in its first `get`, since `last_change_tick` (which, again, was `0`) would be piped directly into any system params and not updated until after the data was fetched once.

Reply:
This change makes sense to me.

Reply:
I think I still have it wrong. Like this, system states initialized "late" wouldn't see any changes. I think to be consistent, the `last_change_tick` should be initialized to `world.change_tick().wrapping_sub(MAX_CHANGE_AGE)` to capture all changes that can be detected.

Reply:
> Having last_change_tick initialized to 0 was probably not correct.

Agreed

> Like this, system states initialized "late" wouldn't see any changes.

I would personally expect them to only detect changes that occurred after they were initialized. Both behaviours have their merits, I think the question should be made very visible in the PR so that reviewers don't miss it.

Reply:
Good point. I'll add it to the description.